docs/contribute/debug-programs.md

```diff
@@ -1,50 +1,67 @@
-# Debugging with GDB
+# Debugging
```

Comment:
Would it be worth including a 'prerequisites' section here and a means  of obtaining anything missing?